projects/kit/components/input-date/input-date.directive.ts

```diff
@@ -118,6 +118,11 @@ export abstract class TuiInputDateBase<
         this.max.set(max instanceof TuiDay ? max : this.options.max);
     }
 
+    public override writeValue(value: T | null): void {
+        super.writeValue(value);
+        this.textfield.value.set(this.stringify(this.value()));
```

Comment:
But that would break the same issue we talked about when person is typing the incomplete value and some state manager pretzel code will continuously push `null` into form control.

Reply:
If we add `if (this.control.pristine) {` check before resetting textfield value to empty string – we will probably cover most of possible cases without breaking the existing functinality.